spring-integration-core/src/main/java/org/springframework/integration/util/MessagingAnnotationUtils.java

```diff
@@ -69,9 +70,31 @@
 		return null;
 	}
 
-	public static boolean hasValue(Object value) {
-		return value != null && (!(value instanceof String) || (StringUtils.hasText((String) value))) // NOSONAR
-				&& (!value.getClass().isArray() || ((Object[]) value).length > 0);
+	/**
+	 * Determines if the value of a named attribute from an annotation instance contains an actual value.
```

Comment:
The method JavaDoc must be imperative, kinda commanding us what to do.
No blank lines in the JavaDoc: https://github.com/spring-projects/spring-framework/wiki/Code-Style#javadoc-formatting